file-updates/test.sh

```diff
@@ -0,0 +1,159 @@
+
+    echo "Creating ReleasePlanAdmission"
+    kubectl apply -f release-resources/release-plan-admission.yaml "$MANAGED_KUBECONFIG"
+
+    echo "Creating EnterpriseContractPolicy"
+    kubectl apply -f release-resources/ec-policy.yaml "$MANAGED_KUBECONFIG"
+
+}
+
+function teardown() {
+
+    kubectl delete pr -l "appstudio.openshift.io/application=$APPLICATION_NAME,pipelines.appstudio.openshift.io/type=build,appstudio.openshift.io/component=$COMPONENT_NAME" "$DEV_KUBECONFIG"
+    kubectl delete pr -l "appstudio.openshift.io/application=$APPLICATION_NAME,pipelines.appstudio.openshift.io/type=release" "$MANAGED_KUBECONFIG"
+    kubectl delete release "$DEV_KUBECONFIG" -o=jsonpath="{.items[?(@.spec.releasePlan==\"$RELEASE_PLAN_NAME\")].metadata.name}"
+    kubectl delete releaseplanadmission "$RELEASE_PLAN_ADMISSION_NAME" "$MANAGED_KUBECONFIG"
+    kubectl delete releasestrategy "$RELEASE_STRATEGY_NAME" "$MANAGED_KUBECONFIG"
+
+    if kubectl get application "$APPLICATION_NAME"  "$DEV_KUBECONFIG" &> /dev/null; then
+        echo "Application $APPLICATION_NAME exists. Deleting..."
+        kubectl delete application "$APPLICATION_NAME" "$DEV_KUBECONFIG"
+    else
+        echo "Application $APPLICATION_NAME does not exist."
+    fi
+}
+
+# Function to watch Build or Release PipelineRun and wait till succeeds.
+function wait_for_pr_to_complete() {
+    local kube_config
+    local type=$1
+    local start_time=$(date +%s)
+
+    if [ "$type" = "release" ]; then
+        kube_config="$MANAGED_KUBECONFIG"
+        crd_labels="appstudio.openshift.io/application=$APPLICATION_NAME,pipelines.appstudio.openshift.io/type=$type"
+    else
+        kube_config="$DEV_KUBECONFIG"
+        crd_labels="appstudio.openshift.io/application=$APPLICATION_NAME,pipelines.appstudio.openshift.io/type=$type,appstudio.openshift.io/component=$COMPONENT_NAME"
+    fi
+
+    while true; do
+        crd_json=$(kubectl get PipelineRun -l "$crd_labels" "$kube_config" -o=json)
+
+        reason=$(echo "$crd_json" | jq -r '.items[0].status.conditions[0].reason')
+        status=$(echo "$crd_json" | jq -r '.items[0].status.conditions[0].status')
+        type=$(echo "$crd_json" | jq -r '.items[0].status.conditions[0].type')
+        name=$(echo "$crd_json" | jq -r '.items[0].metadata.name')
+        namespace=$(echo "$crd_json" | jq -r '.items[0].metadata.namespace')
+
+        if [ "$status" = "False" ] || [ "$type" = "Failed" ]; then
+            echo "PipelineRun $name failed."
+            return 1
+        fi
+
+        if [ "$status" = "True" ] && [ "$type" = "Succeeded" ]; then
+            echo "PipelineRun $name succeeded."
+            return 0
+        else
+            current_time=$(date +%s)
+            elapsed_time=$((current_time - start_time))
+
+            if [ "$elapsed_time" -ge "$TIMEOUT_SECONDS" ] ; then
+                echo "Timeout: PipelineRun $name in namespace $namespace did not succeeded within $TIMEOUT_SECONDS seconds."
+                return 1
```

Comment:
Note that the script does not have `set -e`, so even if this function returns 1, the script will carry on. E.g. if the build times out, the script will move on to wait for the release pipeline. I think it should fail, no? So I would suggest to add `set -e`. But then the tearDown function (especially if we call it before the test) should be modified to be gracefull - e.g. `kubectl delete ... || true`

Reply:
the script is working now...I'd like to defer this to a later iteration.

Reply:
OK, fine with me.